lib/arux_app/api/auth.rb

```diff
@@ -60,6 +60,32 @@ def authorization_url(scope: "public")
         base_uri.to_s
       end
 
+      def basic_authentication(username, password, scope = "public")
+        params = {
+          scope: scope,
+          grant_type: "password",
+          client_id: client_id,
+          client_secret: client_secret
+        }
+        request = HTTPI::Request.new
+        request.url = "#{self.class.server_uri}/oauth/token"
+        request.body = params
+        request.headers = { 'User-Agent' => USER_AGENT }
+        request.auth.basic(username, password)
```

Comment:
I looked at the [source](https://github.com/savonrb/httpi/blob/main/lib/httpi/request.rb#L17-L23) for `HTTPI::Request`, and it looks like you can do this:

```suggestion
        request = HTTPI::Request.new(
          url: "#{self.class.server_uri}/oauth/token",
          body: params,
          headers: { "User-Agent" => USER_AGENT }
      )
      request.auth.basic(username, password)
```

That still leaves `request.auth.basic` that you can't pass in. You could also do:

```rb
        request = HTTPI::Request.new.tap do |req|
          req.url = "#{self.class.server_uri}/oauth/token"
          req.body = params
          req.headers = { 'User-Agent' => USER_AGENT }
          req.auth.basic(username, password)
        end
```

This is a super minor thing, but I like to be able to visually see a chunk of stuff that is all working to build up an object as one big thing, not a series of separate lines, if that makes sense. Just makes it a little more readable for me personally.

Reply:
Good call on `#tap` works well and improves readability like you mentioned.